JavaScript/sign-script.js

```diff
@@ -0,0 +1,77 @@
+// const emailInput = document.getElementsByClassName('email')[0];
+// const errorMessage = document.getElementsByClassName('error-message')[0];
+const emailInput = document.querySelectorAll('.email')[0];
+const passInput = document.querySelectorAll('.password')[0];
+const passConfirmInput = document.querySelectorAll('.password-confirm')[0];
+const errorMail = document.querySelectorAll('.error-mail')[0];
+const errorPass = document.querySelectorAll('.error-pass')[0];
+const errorPassConfirm = document.querySelectorAll('.error-passconfirm')[0];
+const eyeImagePass = document.querySelector('#eyeImagePass'); 
+const eyeImageConfirm = document.querySelector('#eyeImageConfirm'); 
+const pattern = /^[A-Za-z0-9_]+[A-Za-z0-9]*[@]{1}[A-Za-z0-9]+[A-Za-z0-9]*[.]{1}[A-Za-z]{1,3}$/;
+let link = '/folder';
+
+function checkEmail(emailType) {
+  if (emailInput.value === '') {
+    errorMail.textContent = '이메일을 입력해주세요.';
+    emailInput.classList.add('error'); // 에러 클래스 추가
+  } else if (!pattern.test(emailType)) {
+    errorMail.textContent = '올바른 이메일 주소가 아닙니다.';
+    emailInput.classList.add('error'); // 에러 클래스 추가
+  } else if (emailInput.value !== '') {
+    errorMail.textContent = '이메일을 확인해주세요.';
+    emailInput.classList.add('error'); // 에러 클래스 추가
+  } else if (emailInput.value === 'test@codeit.com' && passInput.value === 'codeit101') {
+    location.href = link;
+  } else {
+    errorMail.textContent = '';
+    emailInput.classList.remove('error'); // 에러 클래스 제거
+  }
```

Comment:
```suggestion
  if (emailInput.value === '') {
    errorMail.textContent = '이메일을 입력해주세요.';
    emailInput.classList.add('error'); // 에러 클래스 추가
    return;
  } 
  
  if (!pattern.test(emailType)) {
    errorMail.textContent = '올바른 이메일 주소가 아닙니다.';
    emailInput.classList.add('error'); // 에러 클래스 추가
    return;
  } 
  
  if (emailInput.value !== '') {
    errorMail.textContent = '이메일을 확인해주세요.';
    emailInput.classList.add('error'); // 에러 클래스 추가
    return;
  } 
  
  if (emailInput.value === 'test@codeit.com' && passInput.value === 'codeit101') {
    location.href = link;
    return;
  }
  
  errorMail.textContent = '';
  emailInput.classList.remove('error'); // 에러 클래스 제거
```

early return을 쓰면 조금 더 깔끔하게 사용이 가능합니다. 이렇게 사용하는 이유는 사실 동작 자체는 비슷합니다. (사실 똑같습니다) 다만 이렇게 사용했을 경우 장점이라고 하면 if문의 위치를 다른 곳으로 뺼 수 있습니다. if else if else 형태로 사용된다면 하나의 if문에 같이 종속되어야하는 경우가 생기지만 if문에 early return을 사용하게 된다면 각각의 if문이 떨어져있는 상태라서 사용하기도 조금 편합니다 :) (아직은 이해를 못하실 수는 있습니다...!)

Reply:
와 이렇게 쓸 수 있군요....! 저는 하나의 비슷한 동작들을 꼭 if문으로 한번에 묶어야 한다고 생각했습니다. 그래야 코드가 짧아지기도 하고요.. 그런데 멘토님이 알려주신대로 보니 훨씬 눈에도 잘 들어오고 종속적이지 않아 수정하기도 편할 것 같습니다. 감사합니다!

Reply:
수정해주신 코드 중 38번줄에 있는 코드는 else문을 아예 밖으로 빼버리신 것 같은데 맞나요? 잘 이해되지 않아 여쭈어봅니다.

Reply:
네 맞습니다 early return을 하게되면 else가 없어도 나머지는 맨 밑에 적어두면 else처럼 실행됩니다.